opta/module_processors/byo_cert.py

```diff
@@ -0,0 +1,103 @@
+from typing import TYPE_CHECKING, List, Tuple
+
+from OpenSSL.crypto import (
+    FILETYPE_PEM,
+    X509,
+    Error,
+    PKey,
+    dump_publickey,
+    load_certificate,
+    load_privatekey,
+)
+
+from opta.exceptions import UserErrors
+from opta.module_processors.base import ModuleProcessor
+
+if TYPE_CHECKING:
+    from opta.layer import Layer
+    from opta.module import Module
+
+
+class BYOCert(ModuleProcessor):
```

Comment:
Not a big fan of this validation. This is code we need to maintain PLUS there might be edge cases here which lead to false negatives. Like what happens if I don't have a cert chain?

Reply:
See message above-- unless you're one of the internet heavyhitters this won't be an issue